packages/core/core/src/ResolverRunner.js

```diff
@@ -105,11 +135,14 @@ export default class ResolverRunner {
     let dir = dependency.sourcePath
       ? path.dirname(dependency.sourcePath)
       : '<none>';
-    let err = new Error(
+
+    let err: any = await this.getThrowableDiagnostic(
```

Comment:
Can this error actually be generalised in the ResolverRunner, but what if it isn't?
Should the resolver return/throw a diagnostic with information on why it didn't resolve? This would enable advanced diagnostics like Rome does but how would we do this if there are multiple resolver, return the last one? If the last one is a simple resolver for a project specific syntax the chances are big that one has far less advanced diagnostics to the Parcel one.

Reply:
Well, we can only show as much information as the resolver gave us... Normally, we return null from resolvers when they don't resolve anything and go to the next one, and we should keep doing that. However, I think the DefaultResolver should throw an error in a couple specific cases:

1. User specified `main`, `module`, etc. in package.json that points to a file that doesn't exist (after trying all extensions).
2. User specifies an `alias` that points to a file that doesn't exist.
3. package.json is not parsable as JSON.
4. Fields we read in package.json are not valid (e.g. `main` is a boolean or something).

In these cases, the errors are caused by user code being incorrect. We should surface those so they can be fixed. These could be additional code frames in addition to the one showing where the import came from in the original code.

Reply:
Sounds good, will do the advanced diagnostics with hints in a follow-up. Gonna focus on the basic diagnostics the resolverrunner can create in this PR